init.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+set -e # Tells bash to immediately exit on failure off a command
+
+REFRESH_TOEN=$(echo "$FLOW_REFRESH_TOKEN" | base64 -d)
+ACCESS_TOKEN_REQUEST=${REFRESH_TOEN/\"id\"/\"refresh_token_id\"}
```

Comment:
```suggestion
ACCESS_TOKEN_REQUEST=${REFRESH_TOKEN/\"id\"/\"refresh_token_id\"}
```

Reply:
I honestly don't really understand what this `${REFRESH_TOKEN/\"id\"/\"refresh_token_id\"}` is doing here 🤔
From its usage below I can infer it's creating the data payload to be sent to `generate_access_token`, but how is this a JSON payload at all?

Reply:
This is here because the refresh tokens from `create_refresh_token` have an `id` field:

https://github.com/estuary/flow/blob/2dcb5d9b909d79097ba012f4dde92944922c437f/supabase/migrations/17_refresh_tokens.sql#L39-L42

But `generate_access_token` takes a `refresh_token_id`: 

https://github.com/estuary/flow/blob/master/supabase/migrations/17_refresh_tokens.sql#L60

So we need to replace `"id"` with `"refresh_token_id"`

Reply:
```
$ curl
  -XPOST
  -H "apikey: ey..."
  -H "Authorization: Bearer ey..."
  -H "Content-Type: application/json"
  https://eyrcnmuzzyriypdajwdk.supabase.co/rest/v/rpc/create_refresh_token
  -d '{"multi_use": true, "valid_for": "10 days"}'

{"id" : "09:1f:..", "secret" : "9e.."}

$ curl 
  -XPOST 
  -H "apikey: ey.." 
  -H 'Content-Type: application/json'
  -d '{"refresh_token_id" : "09:1f:..", "secret" : "9e.."}' 
  https://eyrcnmuzzyriypdajwdk.supabase.co/rest/v1/rpc/generate_access_token

{"access_token" : "ey.."}
```

Reply:
Ah! I didn't know bash supports replacing strings out of the box, I always used `sed`. Can you please add a comment about this? Thank you!

Reply:
Lol neither did I until I tried searching for the right `sed` incantation

Reply:
Buuut I decided to just use `jq` instead since it's clearer and possibly less error-prone

```bash
# Replace "id" with "refresh_token_id" because `create_refresh_token` returns
# {"id" : "...", "secret" : "..."}
# But `generate_access_token` expects
# {"refresh_token_id" : "...", "secret" : "..."}
ACCESS_TOKEN_REQUEST=$(echo "$REFRESH_TOKEN" | jq -c -r '. + {"refresh_token_id": .id}| del(.id)')
```

Reply:
Thanks Joseph! I think this has the same output but is simpler:

```
echo "$REFRESH_TOKEN" | jq -c -r '{"refresh_token_id": .id}'
```

Reply:
Good point.. but doesn't it also need the secret?

Reply:
ah yes, sorry, the secret also needs to be there